src/crimson/osd/ops_executer.h

```diff
@@ -305,6 +315,16 @@ class OpsExecuter : public seastar::enable_lw_shared_from_this<OpsExecuter> {
   }
 
   version_t get_last_user_version() const;
+
+  const SnapContext& get_snapc() const {
+    return snapc;
+  }
+
+  void make_writeable(std::vector<pg_log_entry_t>& log_entries);
+
+  object_info_t* prepare_clone(const hobject_t& coid);
```

Comment:
Added.
```
  // For primary osd, the returned value will point to the newly
  // created clone object info. Otherwise, to an on-stack static snap.
  object_info_t* prepare_clone(
    const hobject_t& coid,
    object_info_t& static_snap_oi);
```